tests/by-util/test_ls.rs

```diff
@@ -3522,3 +3522,126 @@ fn test_ls_perm_io_errors() {
         .code_is(1)
         .stderr_contains("Permission denied");
 }
+
+#[test]
+fn test_ls_dired_incompatible() {
+    let scene = TestScenario::new(util_name!());
+
+    scene
+        .ucmd()
+        .arg("--dired")
+        .fails()
+        .code_is(1)
+        .stderr_contains("--dired requires --format=long");
+}
+
+#[test]
+fn test_ls_dired_recursive() {
+    let scene = TestScenario::new(util_name!());
+
+    scene
+        .ucmd()
+        .arg("--dired")
+        .arg("-l")
+        .arg("-R")
+        .succeeds()
+        .stdout_does_not_contain("//DIRED//")
+        .stdout_contains("  total 0")
+        .stdout_contains("//SUBDIRED// 2 3")
+        .stdout_contains("//DIRED-OPTIONS// --quoting-style");
+}
+
+#[test]
+fn test_ls_dired_simple() {
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+    scene
+        .ucmd()
+        .arg("--dired")
+        .arg("-l")
+        .succeeds()
+        .stdout_contains("  total 0");
+
+    at.mkdir("d");
+    at.touch("d/a1");
+    let mut cmd = scene.ucmd();
+    cmd.arg("--dired").arg("-l").arg("d");
+    let result = cmd.succeeds();
+    result.stdout_contains("  total 0");
+    println!("    result.stdout = {:#?}", result.stdout_str());
+
+    let dired_line = result
+        .stdout_str()
+        .lines()
+        .find(|&line| line.starts_with("//DIRED//"))
+        .unwrap();
+    let positions: Vec<usize> = dired_line
+        .split_whitespace()
+        .skip(1)
+        .map(|s| s.parse().unwrap())
+        .collect();
+
+    assert_eq!(positions.len(), 2);
+
+    let start_pos = positions[0];
+    let end_pos = positions[1];
+
+    // Extract the filename using the positions
+    let filename =
+        String::from_utf8(result.stdout_str().as_bytes()[start_pos..=end_pos].to_vec()).unwrap();
+
+    assert_eq!(filename, "a1\n");
+}
+
+#[test]
+fn test_ls_dired_complex() {
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+
+    at.mkdir("d");
+    at.mkdir("d/d");
+    at.touch("d/a1");
+    at.touch("d/a22");
+    at.touch("d/a333");
+    at.touch("d/a4444");
+
+    let mut cmd = scene.ucmd();
+    cmd.arg("--dired").arg("-l").arg("d");
+    let result = cmd.succeeds();
+    // Number of blocks
+    #[cfg(target_os = "linux")]
+    result.stdout_contains("  total 4");
```

Comment:
For some reason this line fails on my machine because the total is 0 :| It works fine, and the total is 4, if I do the steps manually...

Reply:
and you run linux ?

Reply:
yep, I'm running Arch Linux.

Reply:
Here is the debug output:
```
Output:
  total 0
  -rw-r--r-- 1 dho dho    0 Sep 18 08:00 a1
  -rw-r--r-- 1 dho dho    0 Sep 18 08:00 a22
  -rw-r--r-- 1 dho dho    0 Sep 18 08:00 a333
  -rw-r--r-- 1 dho dho    0 Sep 18 08:00 a4444
  drwxr-xr-x 2 dho dho   40 Sep 18 08:00 d
//DIRED// 51 53 95 98 140 144 186 191 233 234
//DIRED-OPTIONS// --quoting-style=literal
```
The two odd things are the timestamps (they're off by two hours) and the size of the directory (it's 40 instead of 4096)

Reply:
fun! 
as you know, it isn't the fault of this PR  :)

Reply:
That's true, I added it to my todo list to investigate further